ui/admin/app/routes/scopes/scope/roles/role/manage-scopes/manage-org-projects.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class ScopesScopeRolesRoleManageScopesManageOrgProjectsRoute extends Route {
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  // =services
+
+  @service store;
+
+  // =methods
+
+  /**
+   * Loads projects for current org scope.
+   * @return {Promise<{role: RoleModel, orgScope: [ScopeModel], projectScopes: [ScopeModel], totalItems: number, totalItemsCount: number, selectedProjectIDs: [string], remainingProjectIDs: [string]}> }
+   */
+  async model({ org_id, search, page, pageSize }) {
+    const role = this.modelFor('scopes.scope.roles.role');
+    const orgScope = this.store.peekRecord('scope', org_id);
+
+    let selectedProjectIDs = role.grantScopeProjectIDs;
+    let remainingProjectIDs = [];
+    if (role.scope.isGlobal) {
+      const projects = await Promise.all(
+        role.grantScopeProjectIDs.map((id) =>
+          this.store.findRecord('scope', id),
```

Comment:
Im rethinking this approach, since it makes an api request and if a role has more than 3k project type grant scopes associated with it then we could run into rate limiting issue. Im thinking instead we query (peek option) into indexedDb for each project grant scope id. Im not considering using `this.store.peekRecord` since on a page refresh the data is no longer in the ember store :/.

Reply:
Oh great point. I think the query should fix that for us.

Reply:
Did we preload all the project scopes ahead of time?

Reply:
yup [here](https://github.com/hashicorp/boundary-ui/blob/llb/multi-scope-grants/ui/admin/app/routes/scopes/scope/roles/role/scopes.js#L16) is where i remember the pre-loading occuring. Tho now that im looking at it closer, im thinking that perhaps that preloading should be pulled out to be on the role route instead of the scopes route because the manage scopes action exists on the role route as well.

Reply:
Ended up moving the pre-loading to the role route and changed the condition for when all scopes should be loaded.